File: sdks/python/src/boxsdk/models/operations/post_shield_information_barrier_segment_members.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import shieldinformationbarrier_base as shared_shieldinformationbarrier_base
from ..shared import user_base as shared_user_base
from ..shared import clienterror as shared_clienterror
from ..shared import shieldinformationbarriersegmentmember as shared_shieldinformationbarriersegmentmember

class PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegmentTypeEnum(str, Enum):
    SHIELD_INFORMATION_BARRIER_SEGMENT = "shield_information_barrier_segment"


@dataclass_json
@dataclasses.dataclass
class PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegment:
    r"""PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegment
    The `type` and `id` of the
    requested shield information barrier segment.
    """
    
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: Optional[PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegmentTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
class PostShieldInformationBarrierSegmentMembersRequestBodyTypeEnum(str, Enum):
    SHIELD_INFORMATION_BARRIER_SEGMENT_MEMBER = "shield_information_barrier_segment_member"


@dataclass_json
@dataclasses.dataclass
class PostShieldInformationBarrierSegmentMembersRequestBody:
    shield_information_barrier_segment: PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegment = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('shield_information_barrier_segment') }})
    user: shared_user_base.UserBase = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('user') }})
    shield_information_barrier: Optional[shared_shieldinformationbarrier_base.ShieldInformationBarrierBase] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('shield_information_barrier') }})
    type: Optional[PostShieldInformationBarrierSegmentMembersRequestBodyTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclasses.dataclass
class PostShieldInformationBarrierSegmentMembersRequest:
    request: Optional[PostShieldInformationBarrierSegmentMembersRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PostShieldInformationBarrierSegmentMembersResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    shield_information_barrier_segment_member: Optional[shared_shieldinformationbarriersegmentmember.ShieldInformationBarrierSegmentMember] = dataclasses.field(default=None)
    

File: sdks/ts/src/sdk/models/operations/postshieldinformationbarriersegmentmembers.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegmentTypeEnum {
    ShieldInformationBarrierSegment = "shield_information_barrier_segment"
}


// PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegment
/** 
 * The `type` and `id` of the
 * requested shield information barrier segment.
**/
export class PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegment extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegmentTypeEnum;
}

export enum PostShieldInformationBarrierSegmentMembersRequestBodyTypeEnum {
    ShieldInformationBarrierSegmentMember = "shield_information_barrier_segment_member"
}


export class PostShieldInformationBarrierSegmentMembersRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=shield_information_barrier" })
  shieldInformationBarrier?: shared.ShieldInformationBarrierBase;

  @SpeakeasyMetadata({ data: "json, name=shield_information_barrier_segment" })
  shieldInformationBarrierSegment: PostShieldInformationBarrierSegmentMembersRequestBodyShieldInformationBarrierSegment;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: PostShieldInformationBarrierSegmentMembersRequestBodyTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=user" })
  user: shared.UserBase;
}


export class PostShieldInformationBarrierSegmentMembersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostShieldInformationBarrierSegmentMembersRequestBody;
}


export class PostShieldInformationBarrierSegmentMembersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  shieldInformationBarrierSegmentMember?: shared.ShieldInformationBarrierSegmentMember;

  @SpeakeasyMetadata()
  statusCode: number;
}
